### PR TITLE
Improve phone numbers for it_IT

### PR DESCRIPTION
**Pull request by FrancYescO**
Implement https://github.com/joke2k/faker/issues/1586

### CODE REVIEW

File: faker/providers/phone_number/it_IT/__init__.py [hunk 1]
```diff
@@ -3,11 +3,194 @@
...
+        "0962",
+        "0963",
+        "0965",
+        "0974",
+        "0975",
+        "099#",
+        "0984",
+        # Mobile: https://it.wikipedia.org/wiki/Prefissi_telefonici_dei_cellulari_italiani
+        "3513", "3514", "3515", "3516", "3517", "3518", "3519", "3520",
+        "330#", "331#", "333#", "334#", "335#", "336#", "337#", "338#", "339#", "360#", "361#", "362#", "363#", "366#", "368#", "381#", "385#",
```
Comment (fcurella) [inline faker/providers/phone_number/it_IT/__init__.py:172]: line too long here https://github.com/joke2k/faker/runs/4696003420?check_suite_focus=true#step:5:15

File: faker/providers/phone_number/it_IT/__init__.py [hunk 2]
```diff
@@ -3,11 +3,194 @@
...
+        "340#", "341#", "342#", "343#", "344#", "345#", "346#", "347#", "348#", "349#", "383#",
+        "320#", "322#", "323#", "324#", "327#", "328#", "329#", "355#", "380#", "388#", "389#", "390#", "391#", "392#", "393#", "397#",
```
Comment (fcurella) [inline faker/providers/phone_number/it_IT/__init__.py:174]: line too long here https://github.com/joke2k/faker/runs/4696003420?check_suite_focus=true#step:5:16

**Review by fcurella — CHANGES_REQUESTED** [review]

**Review by fcurella — CHANGES_REQUESTED** [review]
Could you run your code through the linters? the `Makefile` has a `lint` task that you can run with `make lint`, provided you have `isort`, `black` and `mypy` installed.

**Comment by FrancYescO** [comment]
@fcurella is still something needed?

**Comment by fcurella** [comment]
Looks good! Thank you!

**Pull request merged**